### DISCUSSION TITLE
Update a variable after execution

### DISCUSSION THREAD

**Discussion by LeonardoGentile** [SilentVoid13/Templater#1354] [Help]
I have a complex set of modules to select the template to use and ask the user to fill different variables. One of these steps is to ask for the title of the new note. 
Everything work as expected but If my template include something like this 

```
# <% tp.file.title %>
```
it won't work because at creation time the note is still untitled, its title will be set in one of the aforementioned steps.
So I wanted to invoke the `on_all_templates_executed` hook but not sure how to use it in this case as all the the examples shows to run some operations on post-execution, in this case I just want to update the content of the note.

Is there a way to execute/invoke the above code on post execution?

The alternative is 
- to execute a function that opens the file, add an H1 and add append some content to it
- put the result of tp.file.title into a variable, then update the variable on post execution (I guess this is impossible rn)

**Comment by Zachatoo** [comment]
`on_all_templates_executed` cannot do what you want, there isn't currently a built in functionality for this.

I recommend storing what the user renames the file in a variable and referencing that instead of `tp.file.title`. For example.

```js
<%*
const title = await tp.system.prompt("New filename);
await tp.file.rename(title);
// ... more code
-%>

# <% title %>
```
